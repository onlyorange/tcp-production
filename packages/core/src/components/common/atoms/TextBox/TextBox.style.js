import { css } from 'styled-components';

const textboxStyles = css`
  margin: 0;
  outline: 0;
  line-height: 44px;
  font-size: ${props => props.theme.fonts.fontSize.textbox}px;
  border-radius: 25px;
  background-color: ${props => props.theme.colors.TEXTBOX.BACKGROUND};
  color: ${props => props.theme.colors.TEXTBOX.COLOR};
  border: 1px solid ${props => props.theme.colors.TEXTBOX.BACKGROUND};
  width: calc(100% - 40px);
  padding: 0 20px;
  background-position: left top;
  background-repeat: no-repeat;
  background-size: contain;

  ${props =>
    props.isErrorState ? `border: 1px solid ${props.theme.colors.TEXTBOX.ERROR_BORDER};` : ''};
  ${props =>
    props.isSuccessState ? `border: 1px solid ${props.theme.colors.TEXTBOX.SUCCESS_BORDER};` : ''};

  ${props =>
    props.textIcon === 'icon-email'
      ? `
    background-image: url(/static/images/email.svg);
    padding: 0 20px 0 52px;
    width: calc(100% - 72px);`
      : ''};

  ${props =>
    props.textIcon === 'icon-sms'
      ? `
    background-image: url(/static/images/chat.svg);
    padding: 0 20px 0 52px;
    width: calc(100% - 72px)`
      : ''};

  ${props =>
    props.disabled
      ? `
      background-color: ${props.theme.fieldBackgroundDisabledColor};
      border-color: ${props.theme.fieldBorderDisabledColor};
    `
      : ''};
  &:focus {
    border: 1px solid ${props => props.theme.colors.TEXTBOX.FOCUS_BORDER};
  }
  ${props => (props.inheritedStyles ? props.inheritedStyles : '')};
`;

export default textboxStyles;
