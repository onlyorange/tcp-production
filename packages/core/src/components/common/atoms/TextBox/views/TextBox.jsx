// @flow
import React from 'react';
import type { Node } from 'react';
import withStyles from '../../../hoc/withStyles';
import StyledTextBox from '../TextBox.style';

/**
 * @param {object} props : Props for button
 * @desc This is a Textbox component. The textbox also supports the icons in the front of the textbox.
 * The textIcon = icon-sms and icon-email are currently supported with natural fill.
 * The border styles for success and error are governed by isErrorState and isSuccessState props.
 * The prop disabled determines if the textbox needs to be disabled or not.
 */

type Props = {
  id?: string,
  className: string,
  ariaLabel?: string,
  name?: string,
  type?: string,
  placeholder?: string,
  isErrorState?: boolean,
  isSuccessState?: boolean,
  onChangeHandler?: func,
};

const TextBox = ({
  className,
  id,
  ariaLabel,
  name,
  type,
  placeholder,
  isErrorState,
  isSuccessState,
  onChangeHandler,
}: Props): Node => (
  <input
    id={id}
    aria-label={ariaLabel}
    className={className}
    name={name}
    type={type}
    placeholder={placeholder}
    isErrorState={isErrorState}
    isSuccessState={isSuccessState}
    onChange={onChangeHandler}
  />
);

TextBox.defaultProps = {
  id: '',
  ariaLabel: '',
  name: '',
  type: 'text',
  placeholder: '',
  isErrorState: false,
  isSuccessState: false,
  onChangeHandler: () => {},
};

export default withStyles(TextBox, StyledTextBox);
export { TextBox as TextBoxVanilla };
