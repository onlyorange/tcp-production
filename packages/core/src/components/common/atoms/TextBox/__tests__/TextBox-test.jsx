import React from 'react';
import { shallow } from 'enzyme';
import TextBox from '../views/TextBox';

describe('Textbox component', () => {
  it('renders correctly', () => {
    const props = {
      type: 'text',
      id: 'abcd',
      className: 'asdfasdf',
    };
    const component = shallow(<TextBox {...props} />);
    expect(component).toMatchSnapshot();
  });
});
