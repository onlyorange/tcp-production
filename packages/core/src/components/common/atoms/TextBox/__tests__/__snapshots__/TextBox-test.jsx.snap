// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Textbox component renders correctly 1`] = `ShallowWrapper {}`;
